src/AppBundle/Facebook/ImageFilter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace AppBundle\Facebook;
+
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Imagine\Image\Point;
+use Symfony\Component\Finder\Finder;
+
+class ImageFilter
+{
+    private $imagine;
+
+    private $watermarks;
+
+    public function __construct(ImagineInterface $imagine, string $watermarksPath)
+    {
+        $this->imagine = $imagine;
+
+        $this->watermarks = $this->generateWatermarks($watermarksPath);
+    }
+
+    /**
+     * Applies watermarks on the given picture path
+     * and returns an array of base64 encoded jpeg pictures.
+     */
+    public function applyWatermarks(string $picture): array
+    {
+        $picture = $this->imagine->open($picture);
+
+        $generatedPictures = [];
+
+        foreach ($this->watermarks as $watermark) {
+            $generatedPictures[] = base64_encode($this->addWatermark($watermark, $picture)->get('jpeg'));
+        }
+
+        return $generatedPictures;
```

Comment:
A micro optim of PHP7 is when the array is only initialized in the loop, so you can remove the one line 31 and do instead:
```php
return $generatedPictures ?? [];
```
The compiled code is more performant and may prevent a memory access.